x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_custom_dashboards.ts

```diff
@@ -0,0 +1,229 @@
+/*
```

Comment:
A lot of boilerplate is needed to call the `useTrackedPromise`.  It'd be nice if we had something like the `callApmApi` utils, eg:

```ts
await callApmApi('POST /internal/apm/custom-dashboard', {
            params: {
              query: { customDashboardId: currentDashboard?.id },
              body: {
                dashboardSavedObjectId: newDashboard.value,
                serviceEnvironmentFilterEnabled: serviceFiltersEnabled,
                serviceNameFilterEnabled: serviceFiltersEnabled,
                kuery: `${SERVICE_NAME}: ${serviceName}`,
              },
            },
            signal: null,
          })
```
I'm not sure why we need the abstraction contained in `useTrackedPromise` in infra in general.

Reply:
Yeah, it will be nice to have for sure! I searched for something similar but I couldn't find anything in Infra. We can think about adding something like that in the future. I found only the `useTrackedPromise` used in a similar way and I agree about the boilerplate but I am not sure how to improve it (I tried to reuse some functions where possible (for error handling and response decoding). 
Another similar case for calling an endpoint on button click is in inventory saved views but there we have the `InventoryViewsClient` and there is also boilerplate around the solution (also the usage of react query (example [here](https://github.com/elastic/kibana/blob/6348ab3275c0d1fd79219f62795c73af8e2a4333/x-pack/plugins/observability_solution/infra/public/hooks/use_inventory_views.ts#L45)) can help in this case as well but I tried to not overcomplicate it and keep it similar to the APM solution)

Reply:
you're right. let's not overcomplicate this PR, but I guess we can open a ticket to create a solution similar to APM's?